mypy/messages.py

```diff
@@ -864,6 +864,11 @@ def typeddict_item_name_not_found(self,
         self.fail('\'{}\' is not a valid item name; expected one of {}'.format(
             item_name, format_item_name_list(typ.items.keys())), context)
 
+    def type_arguments_not_allowed(self,
+                                   context: Context,
+                                   ) -> None:
+        self.fail('Type arguments not allowed in this context', context)
```

Comment:
I would prefer a more detailed error message explaining that this is because of ``isinstance``. For example: "Type arguments not allowed in instance and type checks"